### PR TITLE
Release agg context in tests part 3

### PR DESCRIPTION
**Pull request by not-napoleon**
This PR, which is hopefully the last in the aggregations test refactoring saga, enables the fix for `MockBigArrays` which in turn enables circuit breaker tracking for a lot of aggs tests.  Consequentially, it causes a bunch of failures from tests that are doing the wrong thing, so this PR includes fixes for those.

In general, I've tried to favor fixing these tests by standardizing them on calling either `AggregatorTestCase#searchAndReduce` or `AggregatorTestCase#testCase`, which have the correct handling for the `BigArrays` held by `AggregationContext`.  In some cases, this has not been possible, and I've just released the `AggregationContext` in a try block within the test.  That's less than ideal, but there aren't many tests I've needed to do that for.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/RareTermsAggregatorTests.java [hunk 1]
```diff
@@ -248,11 +247,14 @@ public void testUnmapped() throws Exception {
...
                     for (int i = 0; i < fieldNames.length; i++) {
                         RareTermsAggregationBuilder aggregationBuilder = new RareTermsAggregationBuilder("_name").field(fieldNames[i]);
+                        /*
                         Aggregator aggregator = createAggregator(aggregationBuilder, indexSearcher, fieldType1, fieldType2);
                         aggregator.preCollection();
                         indexSearcher.search(new MatchAllDocsQuery(), aggregator.asCollector());
                         aggregator.postCollection();
-                        RareTerms result = (RareTerms) aggregator.buildTopLevel();
+
+                         */
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/RareTermsAggregatorTests.java:256]: leftover commenting?

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxAggregatorTests.java [hunk 2]
```diff
@@ -439,12 +438,16 @@ public void testSingleValuedFieldGetProperty() throws IOException {
...
 
+        /*
         GlobalAggregator aggregator = createAggregator(aggregationBuilder, indexSearcher, fieldType);
         aggregator.preCollection();
         indexSearcher.search(new MatchAllDocsQuery(), aggregator.asCollector());
         aggregator.postCollection();
 
         Global global = (Global) aggregator.buildTopLevel();
+
+         */
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxAggregatorTests.java:449]: Also leftover?

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxAggregatorTests.java [hunk 3]
```diff
@@ -901,13 +901,13 @@ public void testCacheAggregation() throws IOException {
 
         Max max = (Max) aggregator.buildAggregation(0L);
 
+         */
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxAggregatorTests.java:904]: Another leftover?

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregatorTests.java [hunk 4]
```diff
@@ -3077,36 +3080,46 @@ public void testParentFactoryValidation() throws Exception {
                 addToDocument(0, document, createDocument("term-field", "a", "long", 100L));
                 indexWriter.addDocument(document);
             }
+            List<Releasable> releasables = new ArrayList<>();
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregatorTests.java:3083]: I think we don't put anything in the list other than `context`  - I think it's slightly more normal to declare `AggregationContext context = null` and then assign something to it and release the context, null or not.

Comment (nik9000) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregatorTests.java:3083]: Or `try()` it, I guess.

**Review by nik9000 — APPROVED** [review]

**Review by martijnvg — APPROVED** [review]
LGTM

**Pull request merged**